Keep fullscreen state in parent component instead of in MapControl

perjo opened the pull request:
**Related issue(s) and PR(s)**  
This PR closes #1354, #1370 and #1372.

<!-- Include below a description of the changes proposed in the pull request -->

- Fixed fullscreen state getting out of sync between MapControls component and its parents (fixes  #1354 and #1370)
- Fixed fullscreen code in interaction partners and turned fullscreen feature on (fixed #1372)

**Type of change**  
<!-- Please delete options that are not relevant -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
 
**List of changes made**  
<!-- Specify what changes have been made and why -->
1. Moved responsibility to enter fullscreen for a component to the component itself instead of letting the `MapControls` component handle it. `MapControls` simple emits `enterFullscreen`/`exitFullscreen` events. 
2. Made sure the fullscreen state in `MapControl` component is always reflected by fullscreen state of the document. 
3. Removed browser-specific code for all browsers expect Safari (all other implement the Fullscreen API). 
4. Turn fullscreen functionality on for interaction partners. 

**Testing**  
<!-- Please delete options that are not relevant -->
See issue #1354, #1370 and #1372. 

**Further comments**  
<!-- Specify questions or related information -->
This is tested on Chrome 111, Firefox 111 and Safari 15.3. It should be tested on Safari i iOS, but not should be expected to work fully on IPhone, see [compatibility matrices](https://developer.mozilla.org/en-US/docs/Web/API/Fullscreen_API#browser_compatibility).


**Left to do** 

**Definition of Done checklist**  
- [x] My code meets the Acceptance Criteria
- [x] My code follows the [NBIS style guidelines](https://github.com/NBISweden/development-guidelines)
- [x] I have performed a self-review of my own code and commented any hard-to-understand areas
- [x] Tests and lint/format validations are passing
- [x] My changes generate no new warnings

Commit:
Fix: Let MapControl component just emit intended fullscreen changes and let parents call the fullscreen API.

To ensure fullscreen state always reflects the actual full screen state.

Commit:
Fix: Implemented enter/exit fullscreen for 3d viewer

Set css class on 2D viewer as well

Commit:
Fix: Use onfullscreenchange property instead of adding eventlisteners to avoid having
to wait for DOM elements.

perjo committed:
Prettified Code!

Commit:
Feature: Disable fullscreen option for interaction partners as unclear if requested

Commit:
Merge branch 'fix/fullscreen-handled-outside-control-component' of github.com:MetabolicAtlas/MetabolicAtlas into fix/fullscreen-handled-outside-control-component

perjo committed:
Merge branch 'main' into fix/fullscreen-handled-outside-control-component

Commit:
Fix: Add support for Safari 15.6 and Safari on iOS

perjo committed:
Prettified Code!

Commit:
Feature: Enable fullscreen functionality for interaction partners

Commit:
Merge branch 'fix/fullscreen-handled-outside-control-component' of github.com:MetabolicAtlas/MetabolicAtlas into fix/fullscreen-handled-outside-control-component

Commit:
Fix: Word list

perjo committed:
Merge branch 'main' into fix/fullscreen-handled-outside-control-component

KattisLej reviewed (approved):
Very well done, looks good

Commit:
Merge branch 'main' into fix/fullscreen-handled-outside-control-component

perjo commented:
So, I looked in to covering these bugs with a test. After struggling a bit it, I found out that it is not that easy to use the Fullscreen API in cypress tests. The reason for this is that browsers only allow fullscreen to entered as a result of a user gesture (you should be able to set a page in fullscreen if the user does not want to, just like are not allowed to pop up a lot of new browser windows). Sadly, cypress is not yet able to bypass this. See [this issue](https://github.com/cypress-io/cypress/issues/1213) for more information. Maybe [cypress-real-events](https://github.com/dmtrKovalenko/cypress-real-events) could library could be used but not sure we wanted a dependency on this lib and it would only work for chromium browsers.

Commit:
Test: Component test for MapControls

Turned buttons in MapControls component into real button elements
instead of spans for accessibility and testability.

Commit:
Format: Prettier

perjo committed:
Prettified Code!

Commit:
Merge branch 'main' into fix/fullscreen-handled-outside-control-component

perjo commented:
Wrote a component test for the `MapControls` test instead. Not sure that it bring very much value but I think is good to explore component tests. It seems suboptimal to write e2e-tests when it is only the frontend code that is tested. But I guess this will have to be made more clear in https://github.com/MetabolicAtlas/MetabolicAtlas/issues/1083 .

As a bonus, the buttons implemented as `span`s now are real `button`s which enables keyboard navigation and improves accessibility overall.

Commit:
build: solved linting violations in cypress test

perjo commented on frontend/cypress/support/component-index.html:
```diff
@@ -0,0 +1,12 @@
+  <body>
+    <div data-cy-root></div>
+  </body>
+</html>
```
This file was generated by cypress, see [docs](https://docs.cypress.io/guides/component-testing/component-framework-configuration/#Custom-Index-File).

dbampalikis commented:
Currently not sure if the issues fixed here are still relevant. Changing it to draft and will pick it up if the bugs come up again
